addons/product/models/product_template.py

```diff
@@ -827,9 +827,16 @@ def _is_combination_possible(self, combination, parent_combination=None):
             # combination has different values than the ones configured on the template
             return False
 
-        if not self.has_dynamic_attributes() and not self._get_variant_for_combination(combination):
-            # the variant has been deleted
-            return False
+        variant = self._get_variant_for_combination(combination)
+
+        if self.has_dynamic_attributes():
+            if variant and not variant.active:
+                # dynamic and the variant has been archived
+                return False
+        else:
+            if not variant or not variant.active:
```

Comment:
Condition feels strange to read ("if variant is falsy check that variant.active is falsy too ..."), this either crashes or `not variant` part is useless.

Why not writing the condition as one condition like before ?

Reply:
The `not variant` part is useless because the ORM returns `False` when accessing a field on an empty record set.

But even if the ORM didn't do that it wouldn't crash since it's an `or`, not an `and`.

I wanted to write the condition this way because this is the business logic of it: return False if there is no variant, or return False if there is a variant but it is not active.

Reply:
My bad for the `or`, still... your "business logic sentence" could be "return False if there is no active variant" which is a shorter pure synonyme of yours, and makes the `not variant` part useless ^^

Anyways, I will not fight for this :) I personally don't like condition to be broken in multiple part like that, an if/else structure which leads to return False or nothing... I prefer a simple if which can be read as a pure "code-translation" of the "sentence" condition "if it has dynamic attributes and a variant but not active, or if there is no active variant, return false"